eliot/tests/corotests.py

```diff
@@ -149,12 +144,44 @@ def test_coroutine_vs_main_thread_context(self):
             current_context,
             [("main", "B"), ("coro", None), ("coro", "A"), ("main", "B")])
 
+    def test_coroutine_vs_main_thread_context_different_thread(self):
+        """
+        A coroutine has a different Eliot context than the thread that runs the
+        event loop, when run in different thread than the one where the context
+        was created.
+        """
+        # Create context in one thread:
+        ctx = AsyncioExecutionContext()
+        current_context = []
+
+        # Run asyncio event loop and coroutines in a different thread:
+        def run():
+            event_loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(event_loop)
+
+            async def coro():
+                current_context.append(("coro", ctx.current()))
+                ctx.push("A")
+                current_context.append(("coro", ctx.current()))
+
+            ctx.push("B")
+            current_context.append(("main", ctx.current()))
+            run_coroutines(coro)
```

Comment:
I guess this is a (asyncio Task/coroutine?) context switch, kind of.  But are there any more context switching cases you want to cover?  It seems like having a switc inside of `coro` would be an interesting case.